lib/replaceRef.js

```diff
@@ -14,8 +14,20 @@ module.exports = replaceRef;
 * @param  {string} basePath - The root path where any relative schema file references can be resolved
 * @return {object} The schema object with all schema file referenced replaced with the actual file content.
 */
+var idmap = {}
 function replaceRef(schema, basePath) {
     var ref = schema.$ref;
+    // Look through top level schema definitions and store them by id
+    if (schema.hasOwnProperty('definitions') && Object.getOwnPropertyNames(idmap).length == 0) {
+        for (var def in schema.definitions) {
+           idmap[schema.definitions[def].id] = schema.definitions[def]
```

Comment:
Looking at the [schema reference](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.5.7), there doesn't seem to be a _requirement_ for there to be an `id` element for `definition`s.